### PR TITLE
PARQUET-1882: [C++] Buffered Reads should allow for 0 length

### PR DESCRIPTION
**Pull request by emkornfield**
Also, there doessn't seem like a great reason to crash when non-zero.

### CODE REVIEW

**Comment by emkornfield** [comment]
CC @pitrou

**Comment by kiszk** [comment]
Looks good to me

File: cpp/src/parquet/file_serialize_test.cc [hunk 1]
```diff
@@ -398,6 +400,48 @@ TEST(TestBufferedRowGroupWriter, MultiPageDisabledDictionary) {
...
+  auto fileReader = ParquetFileReader::Open(source, props);
+  auto rowGroupReader = fileReader->RowGroup(0);
+  auto columnReader = std::static_pointer_cast<Int32Reader>(rowGroupReader->Column(0));
+
+  int64_t valuesRead;
+  defLevels[0] = -1;
+  defLevels[1] = -1;
+  defLevels[2] = -1;
+  columnReader->ReadBatch(3, defLevels, nullptr, values, &valuesRead);
+  EXPECT_THAT(defLevels, ElementsAre(0, 0, 0));
```
Comment (wesm) [inline cpp/src/parquet/file_serialize_test.cc:442]: Should de-camel-case this test

**Review by wesm — COMMENTED** [review]
Looks good aside from style issues

File: cpp/src/parquet/file_serialize_test.cc (hunk 1 above)
Comment (emkornfield) [reply cpp/src/parquet/file_serialize_test.cc:442]: yeah, sorry, more or less copied and pasted from JIRA

Comment (emkornfield) [reply cpp/src/parquet/file_serialize_test.cc]: done.

**Pull request closed without merging**